edb/server/protocol/protocol.pyx

```diff
@@ -865,43 +865,65 @@ cdef class HttpProtocol:
             username, opt_password = auth_helpers.extract_http_user(
                 scheme, auth_payload, request.params)
 
-            # Fetch the configured auth method
-            authmethod = await self.tenant.get_auth_method(
+            # Fetch the configured auth methods
+            authmethods = await self.tenant.get_auth_methods(
                 username, srvargs.ServerConnTransport.SIMPLE_HTTP)
-            authmethod_name = authmethod._tspec.name.split('::')[1]
 
-            # If the auth method and the provided auth information match,
-            # try to resolve the authentication.
-            if authmethod_name == 'JWT' and scheme == 'bearer':
-                if not self.is_tls:
-                    raise errors.AuthenticationError(
-                        'JWT HTTP auth must use HTTPS')
+            auth_errors = {}
+
+            for authmethod in authmethods:
+                authmethod_name = authmethod._tspec.name.split('::')[1]
+                try:
+                    # If the auth method and the provided auth information
+                    # match, try to resolve the authentication.
+                    if authmethod_name == 'JWT' and scheme == 'bearer':
+                        if not self.is_tls:
+                            raise errors.AuthenticationError(
+                                'JWT HTTP auth must use HTTPS')
```

Comment:
We talked about dropping this also, right?
Do you want to do that separately?

Reply:
Separately